genesyscloud/sdk_client_pool.go

```diff
@@ -0,0 +1,121 @@
+package genesyscloud
+
+import (
+	"context"
+	"log"
+	"sync"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/platformclientv2"
+)
+
+// SDKClientPool holds a pool of client configs for the Genesys Cloud SDK. One should be
+// acquired at the beginning of any resource operation and released on completion.
+// This has the benefit of ensuring we don't issue too many concurrent requests and also
+// increases throughput as each token will have its own rate limit.
+type SDKClientPool struct {
+	pool chan *platformclientv2.Configuration
```

Comment:
Did you consider just using a sync.Pool?

Reply:
> Did you consider just using a sync.Pool?

I looked at it, but it didn't appear to have a way to set a size limit.